### PR TITLE
Add new functions to manage user groups

### PR DESCRIPTION
**Pull request by markphelan**

### CODE REVIEW

File: Atlassian.Bitbucket.psd1 [hunk 1]
```diff
@@ -12,7 +12,7 @@
     RootModule        = 'Atlassian.Bitbucket.psm1'
 
     # Version number of this module.
-    ModuleVersion     = '0.25.1'
+    ModuleVersion     = '0.25.2'
```
Comment (eelkram) [inline Atlassian.Bitbucket.psd1:15]: This would be considered a minor bump as it adds new functionality so 0.26.0

File: Atlassian.Bitbucket.psd1 [hunk 2]
```diff
@@ -108,6 +109,7 @@
         'Get-BitbucketTeam',
         'Get-BitbucketUser',
         'Get-BitbucketUsersByGroup',
+        'Get-BitbucketUserGroup',
```
Comment (eelkram) [inline Atlassian.Bitbucket.psd1:112]: Should we call this Get-BitbucketGroup?

File: Atlassian.Bitbucket.Repository.BranchRestriction.psm1 [hunk 3]
```diff
@@ -167,6 +167,8 @@ function New-BitbucketRepositoryBranchRestrictionMergeCheck {
 }
 
 function ConvertTo-BranchRestriction {
+    [OutputType([MergeCheck])]
```
Comment (eelkram) [inline Atlassian.Bitbucket.Repository.BranchRestriction.psm1:170]: Should the output type just be BranchRestriction?

**Review by eelkram — CHANGES_REQUESTED** [review]
Thanks for submitting the PR.  I have a few small comments if you can take a look.  I assume you have tested this locally and confirmed it works as expected.

**Comment by eelkram** [comment]
Also, please add a changelog entry.

File: Atlassian.Bitbucket.psd1 (hunk 2 above)
Comment (markphelan) [reply Atlassian.Bitbucket.psd1]: I was unsure about this as I wrote it to align with "User" functionality and match the label in the web UI, but happy to match the API endpoint path instead and refer to it as Group.

File: Atlassian.Bitbucket.Repository.BranchRestriction.psm1 (hunk 3 above)
Comment (markphelan) [reply Atlassian.Bitbucket.Repository.BranchRestriction.psm1:170]: This fails running the Pester checks for me if the OutputType is omitted or does not include MergeCheck:
```
Context Atlassian.Bitbucket.Repository.BranchRestriction.psm1
      [-] On line: 181 - The cmdlet 'ConvertTo-BranchRestriction' returns an object of type 'MergeCheck' but this type is not declared in the OutputType attribute. 23ms
        Expected $false, but got $true.
        25:                                     $true | Should Be $False
        at <ScriptBlock>, C:\Users\mark.phelan\repos\Atlassian.Bitbucket\Tests\1.ScriptAnalyzer.Tests\1.ScriptAnalyzer.Tests.ps1: line 25
```

**Comment by markphelan** [comment]
> I assume you have tested this locally and confirmed it works as expected.

Yes, used the new functionality to add a user to a group within a workspace I manage.

**Review by eelkram — APPROVED** [review]
Looks good, thanks for the updates!

**Pull request merged**